#167330969 User view specific property from database

mucyomiller opened the pull request:
#### What does this PR do?
User View Specific property from the database
#### Description of Task to be completed?
- add endpoint /api/v2/property/:id for viewing specific property from database
- add fixes for old tests to fails gracefully while we are not yet updated to v2
- add refactoring everywhere we have v1 -> v2

#### How should this be manually tested?
1) review  source codes
2) use HTTP client hit /api/v2/property/:id endpoint

#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
#167330969
#### Screenshots (if appropriate)
N/A
#### Questions:
N/A

mucyomiller committed:
feature(DB): user view specific property from database
- add end point /api/v2/property/:id for viewing specific property from database
- add fixes for old tests to fails gracefully while we are not yet update to v2
- add refactoring everywhere we have v1 -> v2
[finishes #167330969]

njeri-ngigi commented on server/controllers/PropertyController.js:
```diff
@@ -15,18 +16,35 @@ cloudinary.config({
+  static async viewAllProperties(req, res) {
+    const { error, response: result } = await queryAll('properties');
+    if (error) {
+      return response(res, 404, error, true);
```
I don't think this is a 404 error. It's not that a resource doesn't exist, but rather the server crashed for a number of reasons. I would rather you return an error 500.

njeri-ngigi commented on server/controllers/PropertyController.js:
```diff
@@ -15,18 +16,35 @@ cloudinary.config({
-      return response(res, 200, property);
+    const { error, response: result } = await find('properties', 'id', parseInt(id, 10));
+    if (error) {
+      return response(res, 200, error, true);
```
If there's an error, I don't think you want to return status 200. Find an appropriate status code for it.

njeri-ngigi commented on server/controllers/PropertyController.js:
```diff
@@ -15,18 +16,35 @@ cloudinary.config({
+    }
+    const { rows, rowCount } = result;
+    if (rowCount > 0) {
+      const [firstItem] = rows; // get first return item
```
good use of array destructuring

njeri-ngigi reviewed (commented):
Other than the comments I've left below the PR looks good. Once you correct them you can merge

mucyomiller committed:
feature(BD): adds quick fixes from reviews
- add also fixes for tests that was failing time to time caused by db_init from package.json
- add also fix to change DB_URI -> DATABASE_URL so that we don't change any thing when we push to heroku
[fixes #167330969]
